### PR TITLE
Add backup function with folder under git ignore

### PR DESCRIPTION
**Pull request by DanielDSSim**
The function is automatically called when the saveAddressBook function is called.

### CODE REVIEW

**Comment by WeeSooJun** [comment]
Looks fine now

**Comment by DanielDSSim** [comment]
Awesome! I let you merge haha.

**Comment by WeeSooJun** [comment]
Hold on ah, have you tried testing if it really saves on a backup? Like made tests to see if the code really backs up?

**Comment by DanielDSSim** [comment]
Yup yup. It's quite magical. Actually creates a new folder hahahaha. I guess that's what Java Paths API does.

**Comment by WeeSooJun** [comment]
Oh wow nice. So let's say I make edits to the json file, is the backup written in as well? Cause what if the backupFilePath always == null. Will that happen?

**Comment by DanielDSSim** [comment]
Uhh its like another copy of the addressbook is saved somewhere else whenever you issue do something like add, edit delete etc. The backupfilepath is related to the orginal filepath which is in data\addressbook.json. Basically it creates backup\data\addressbook.json.

**Comment by WeeSooJun** [comment]
Is there any JUnit test or did you try it out and it works? Just confirming hahaha

**Comment by DanielDSSim** [comment]
I tried out manually. Have not tried JUnit tests yet.

**Comment by WeeSooJun** [comment]
Hmmm okay, maybe just add the test later when you have time.

**Comment by DanielDSSim** [comment]
Okay added a JUnit test

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 1]
```diff
@@ -62,6 +62,14 @@ public void addressBookReadSave() throws Exception {
         assertEquals(original, new AddressBook(retrieved));
     }
 
+    @Test
+    public void addressBookReadBackup() throws Exception {
+        AddressBook original = getTypicalAddressBook();
+        storageManager.backupAddressBook(original);
+        ReadOnlyAddressBook retrieved = storageManager.readAddressBook().get();
```
[the remark this answers is not in the document]
Comment (DanielDSSim) [reply src/test/java/seedu/address/storage/StorageManagerTest.java]: Will refine at a later date

**Pull request merged**